### PR TITLE
[New Scheduler] Add a centralized watcher for etcd data

### PR DESCRIPTION
**Pull request by jiangpengcheng**
## Description

The FPC scheduler will use etcd to store and share some data, and some components need to get noticed for some data when they are changed(inserted, updated or removed), this PR implements a centralized watcher for them, which will watch all changes on etcd and distribute them to subscribers

design document: https://cwiki.apache.org/confluence/display/OPENWHISK/WatcherService

## Related issue and scope

- [ ] I opened an issue to propose and discuss this change (#????)

## My changes affect the following components

- [ ] API
- [ ] Controller
- [ ] Message Bus (e.g., Kafka)
- [ ] Loadbalancer
- [x] Scheduler
- [ ] Invoker
- [ ] Intrinsic actions (e.g., sequences, conductors)
- [ ] Data stores (e.g., CouchDB)
- [x] Tests
- [ ] Deployment
- [ ] CLI
- [ ] General tooling
- [ ] Documentation

## Types of changes

- [ ] Bug fix (generally a non-breaking change which closes an issue).
- [x] Enhancement or new feature (adds new functionality).
- [ ] Breaking change (a bug fix or enhancement which changes existing behavior).

## Checklist:

- [x] I signed an [Apache CLA](https://github.com/apache/openwhisk/blob/master/CONTRIBUTING.md).
- [x] I reviewed the [style guides](https://github.com/apache/openwhisk/wiki/Contributing:-Git-guidelines#code-readiness) and followed the recommendations (Travis CI will check :).
- [x] I added tests to cover my changes.
- [ ] My changes require further changes to the documentation.
- [ ] I updated the documentation where necessary.

### CODE REVIEW

File: common/scala/src/main/scala/org/apache/openwhisk/core/service/WatcherService.scala [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+import scala.collection.JavaConverters._
+import scala.collection.concurrent.TrieMap
+
+// messages received by this actor
+case class WatchEndpoint(key: String,
+                         value: String,
+                         isPrefix: Boolean,
+                         name: String,
+                         listenEvents: Set[EtcdEvent] = Set.empty)
+case class UnWatchEndpoint(watchKey: String, isPrefix: Boolean, watchName: String, needFeedback: Boolean = false)
```
Comment (style95) [inline common/scala/src/main/scala/org/apache/openwhisk/core/service/WatcherService.scala:35]:
```suggestion
case class UnwatchEndpoint(watchKey: String, isPrefix: Boolean, watchName: String, needFeedback: Boolean = false)
```

**Review by style95 — COMMENTED** [review]
This is given from https://github.com/apache/openwhisk/pull/5063

**Comment by ningyougang** [comment]
LGTM

**Review by ningyougang — APPROVED** [review]

**Comment by KeonHee** [comment]
Referring to review https://github.com/apache/openwhisk/pull/5068#pullrequestreview-591021999, you need to add `org/apache/openwhisk/core/service/**` to `REQUIRE_SCHEDULER` to run the test in CI .

**Comment by bdoyle0182** [comment]
LGTM

**Review by KeonHee — APPROVED** [review]

**Pull request merged**